charts/shoot-core/charts/csi-packet/Chart.yaml

```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
```

Comment:
I think I grasp what you are doing there, but not 100% sure. Can we let this go for now, as this is a large and complex PR, and then do a separate one migrating the relevant CSI parts from seed to shoot after this is in? That will make for a much smaller (and easier to understand) PR, and give me the opportunity to ask questions relevant just to that.

Reply:
Packet support is not yet in, so you have to add the CSI manifests anyway. Currently, you are adding them to `shoot-core` chart which would mean that all CSI-related components are deployed into the shoot. I know that you adapted this from the Alicloud PR, however, it was done wrongly there in the first place. With #928 this has been fixed.
Now, only the provider-specific disk driver is deployed to the shoot (see https://github.com/gardener/gardener/tree/master/charts/shoot-core/charts/csi-alicloud) while all the CSI controllers are deployed to the seed (see https://github.com/gardener/gardener/tree/master/charts/seed-controlplane/charts/csi-alicloud).

Maybe, instead of looking at #928, get yourself some inspiration from the `master` branch itself.
IMHO it's not more complex or effort to do it right already now. You haven't added any CSI manifests/charts yet anyways, so whether you deploy them to shoot or to seed doesn't really matter. However, when choosing the first option (which is not even fully working when shoot clusters are hibernated (because then the volume controllers don't run)) you have additional and unneeded migration effort afterwards.

Reply:
OK, going to see if I can get this.

Reply:
I am unclear as to what should go where. It is clear that what used to be in `shoot-core` is now split between `shoot-core` and `seed-controlplane`. 

Here are all of the parts for CSI as defined in the templates:

* `service/csi-packet-cd` - fronts the `StatefulSet` (next line)
* `statefulset/csi-packet-controller` - runs the containers for the following:
    * `csi-provisioner`
    * `csi-attacher`
    * `packet-driver`
* `secret/packet-cloud-config` - contains API key and project ID
* `daemonset/csi-node` - runs the containers for the following:
    * `csi-driver-registrar`
    * `packet-driver`
* a bunch of `ServiceAccount` / `ClusterRole` / `ClusterRoleBinding`

What should go where?

Reply:
I cannot tell you that exactly because I don't know what these components are doing. Generally, everything that is related to the CSI controllers themselves should go to the `seed-controlplane` chart. Everything that is related to the CSI driver plugin (that must run on the worker nodes) has to go to `shoot-core`.

Reply:
I think I can work with that. I think the hard part for me to grasp is the idea of splitting it. Normally, I would just deploy it into a k8s cluster, and k8s takes care of deploying things where they need to go: StatefulSets wherever, DaemonSets to every node, Service as a concept, etc. Now, I need to think about how to split them. If there are shared dependencies, I would need to break them out. It makes it harder to work with.

I am willing to try, but what is the real value in doing it this way?

Reply:
OK, I figured it out. Coming shortly.

Reply:
I did figure it out. Got the control plane stuff into `seed-controlplane` along with its various service account, clusterrole, rolebinding, etc.; and got the per-node stuff into `shoot-core`, along with its various service account, clusterrole, rolebinding, etc.

The only question that remains here is about the secrets, notably `apiToken` and `projectID`. 

I got from before that these are injected into the `seed-controlplane` as the secret `cloudprovider`. However, the alicloud example injects them as a separate secret in a separate `.yaml` for `shoot-core`.  I did the same here. Please take a look.

Is that correct?

Reply:
I have put some comments to the files. **In the seed**, Gardener creates the `cloudprovider` secret that contains the `apiToken` and the `projectID`. In the shoot, Gardener does not automatically create this secret. That's why the alicloud CSI chart in `shoot-core` has https://github.com/gardener/gardener/blob/master/charts/shoot-core/charts/csi-alicloud/templates/csi-diskplugin-secret.yaml.